pkg/discovery/helper.go

```diff
@@ -117,6 +119,7 @@ func (h *helper) ResourceFor(input schema.GroupVersionResource) (schema.GroupVer
 	return gvr, apiResource, nil
 }
 
+// KindFor expects input where kind is lowercased to hit the map
```

Comment:
The reason the `Kind` doesn't match is that the test code converts the `Resource` to `Kind` incorrectly: https://github.com/vmware-tanzu/velero/blob/main/pkg/test/discovery_client.go#L79

This shouldn't happen in the real world.

So let's specify the `Kind` explicitly in the test code, I create a [PR](https://github.com/vmware-tanzu/velero/pull/7334) for this, free feel to copy the changes into your PR if it isn't merged